src/steamship/data/plugin/index_plugin_instance.py

```diff
@@ -73,10 +76,19 @@ class SearchResults(CamelModel):
     items: List[SearchResult] = None
 
     @staticmethod
-    def from_query_results(query_results: QueryResults) -> "SearchResults":
-        items = [SearchResult.from_query_result(qr) for qr in query_results.items or []]
+    def from_query_results(query_results: QueryResults, client: Client) -> "SearchResults":
+        items = [
+            SearchResult.from_query_result(qr, client=client) for qr in query_results.items or []
+        ]
         return SearchResults(items=items)
 
+    def to_ranked_blocks(self) -> List[Block]:
+        blocks = []
+        for result in self.items:
+            tag = result.tag
+            blocks.append(Block.get(tag.client, tag.block_id))
```

Comment:
if `get()` raises an exception, should the entire operation fail?  do we want a graceful handling?